MPRIS & global media hotkeys broken in Gnome in 4.7.0-pre builds

<!---
Hey There!
Thanks for taking the time to raise an issue.
Before you click that big tempting "Submit New Issue" button though, can you quickly run through the list below.
-->

**Checklist:**

* [x] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [x] I have double checked and can reproduce the issue

<!--

If you have done those 3 things go ahead and fill out the information below.
Once again, thanks for taking the time to submit an issue

If appropriate please attach the DEBUG information zip file that you can generate by following the instructions in the FAQ  
(https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ)

-->

**OS:** Linux -- Debian Buster

**GPMDP Version:** CI build of 8f86ba735f7afe7d9a72a40eda8c4b24fbc42a97 (also happens with some earlier 4.7.0-pre builds)

**Issue Descriptions:**
The media player interface that used to integrate with Gnome no longer works in the current CI builds.  It works fine in the last 4.6.x release build.  Gnome doesn't think any media player app is running at all, and as such shows no panel indicator, nor do global media hotkeys work.

The merge of #3513 is immediately suspect in the commit history, but I have not bisected to prove that theory.  Given that #3510 claims that a similar issue on MacOS was _fixed_ via the Electron update, I suppose that may be suspect as being related on Linux?

**Steps to Reproduce:**
1. Install GPMDP on a Gnome system
2. Start playing music
3. Look in the system menu for a media player indicator -- nothing
4. Try using global media hotkeys bound via the gnome keyboard settings -- they show an indicator that nothing is playing

See also:
* https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3548#issuecomment-486929862 and following related comments

OK, I did some UTSL & figured out how to launch dev mode from the CLI, and now I have what is probably a much more useful error trace for you :)

```
error: Uncaught Exception. Error: Cannot find module 'xmlbuilder'
    at Module._resolveFilename (internal/modules/cjs/loader.js:602:15)
    at Function.Module._resolveFilename (/usr/share/google-play-music-desktop-player/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (internal/modules/cjs/loader.js:528:25)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/builder.js:7:13)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/builder.js:127:4)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/builder.js:129:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/xml2js.js:10:13)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/xml2js.js:37:4)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/xml2js/lib/xml2js.js:39:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/lib/service/name.js:6:16)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/lib/service/name.js:171:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/lib/bus.js:5:14)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/lib/bus.js:351:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/index.js:2:20)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/dbus-next/index.js:144:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/linux/mediaKeysDBus.js:11:17)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/linux/mediaKeysDBus.js:49:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/linux/index.js:11:1)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/linux/index.js:14:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/index.js:13:1)
    at Object.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/main/features/index.js:14:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at App.<anonymous> (/usr/share/google-play-music-desktop-player/resources/app.asar/build/index.js:187:5)
    at App.emit (events.js:187:15)
```

--- a/package.json
+++ b/package.json
@@ -160,13 +160,13 @@
     "@nodert-win10/windows.media.playback": "^0.2.95",
     "@nodert-win10/windows.storage.streams": "^0.2.95",
     "appdmg": "^0.4.5",
-    "dbus-next": "^0.6.1",
+    "dbus-next": "^0.7.1",
     "electron-installer-debian": "^0.4.0",
     "electron-installer-redhat": "^0.3.0",
     "electron-media-service": "^0.2.2",
     "ll-keyboard-hook-win": "^3.0.0",
     "mouse-forward-back": "^1.0.1",
-    "mpris-service": "^2.0.0"
+    "mpris-service": "^2.1.0"
   },
   "babel": {
     "presets": [

--- a/src/main/features/linux/mediaKeysDBus.js
+++ b/src/main/features/linux/mediaKeysDBus.js
@@ -1,23 +1,30 @@
 import dbus from 'dbus-next';
+import { app } from 'electron';
 
-function registerBindings(desktopEnv, session) {
-  const listener = (err, iface) => {
-    if (!err) {
-      iface.on('MediaPlayerKeyPressed', (n, keyName) => {
-        switch (keyName) {
-          case 'Next': Emitter.sendToGooglePlayMusic('playback:nextTrack'); return;
-          case 'Previous': Emitter.sendToGooglePlayMusic('playback:previousTrack'); return;
-          case 'Play': Emitter.sendToGooglePlayMusic('playback:playPause'); return;
-          case 'Stop': Emitter.sendToGooglePlayMusic('playback:stop'); return;
-          default: return;
-        }
-      });
-      iface.GrabMediaPlayerKeys('GPMDP', 0); // eslint-disable-line
+async function registerBindings(desktopEnv, session) {
+  const listener = (n, keyName) => {
+    switch (keyName) {
+      case 'Next': Emitter.sendToGooglePlayMusic('playback:nextTrack'); return;
+      case 'Previous': Emitter.sendToGooglePlayMusic('playback:previousTrack'); return;
+      case 'Play': Emitter.sendToGooglePlayMusic('playback:playPause'); return;
+      case 'Stop': Emitter.sendToGooglePlayMusic('playback:stop'); return;
+      default: return;
     }
   };
 
-  session.getInterface(`org.${desktopEnv}.SettingsDaemon`, `/org/${desktopEnv}/SettingsDaemon/MediaKeys`, `org.${desktopEnv}.SettingsDaemon.MediaKeys`, listener);
-  session.getInterface(`org.${desktopEnv}.SettingsDaemon.MediaKeys`, `/org/${desktopEnv}/SettingsDaemon/MediaKeys`, `org.${desktopEnv}.SettingsDaemon.MediaKeys`, listener);
+  const legacy = await session.getProxyObject(`org.${desktopEnv}.SettingsDaemon`, `/org/${desktopEnv}/SettingsDaemon/MediaKeys`);
+  legacy.getInterface(`org.${desktopEnv}.SettingsDaemon.MediaKeys`);
+  legacy.on('MediaPlayerKeyPressed', listener);
+  app.on('browser-window-focus', () => {
+    legacy.GrabMediaPlayerKeys('GPMDP', 0); // eslint-disable-line
+  });
+
+  const future = await session.getProxyObject(`org.${desktopEnv}.SettingsDaemon.MediaKeys`, `/org/${desktopEnv}/SettingsDaemon/MediaKeys`);
+  future.getInterface(`org.${desktopEnv}.SettingsDaemon.MediaKeys`);
+  future.on('MediaPlayerKeyPressed', listener);
+  app.on('browser-window-focus', () => {
+    future.GrabMediaPlayerKeys('GPMDP', 0); // eslint-disable-line
+  });
 }
 
 try {
@@ -25,5 +32,5 @@ try {
   registerBindings('gnome', session);
   registerBindings('mate', session);
 } catch (e) {
-  // do nothing.
+  // Do nothing
 }

